12_glob_lab/globlab.txt

```diff
@@ -3,27 +3,36 @@ https://www.tocode.co.il/bundles/unix1/lessons/globlab
 
 1. Type in a glob pattern to match:
 
-  1. All files whose name starts with a lowercase letter
+  
+1. All files whose name starts with a lowercase letter
   Answer: 
 
-  2. All files whose name starts with an uppercase letter
+  echo [a-z]*
+2. All files whose name starts with an uppercase letter
   Answer:
 
-  3. All files whose name ends in an "a", "e", "i", "o" or "u"
+  LC_ALL=C bash -c 'echo [A-Z]*'
+3. All files whose name ends in an "a", "e", "i", "o" or "u"
   Answer:
 
-  4. All files whose name DOES NOT end in an "a", "e", "i", "o" or "u"
+  echo *[aeiou]
+4. All files whose name DOES NOT end in an "a", "e", "i", "o" or "u"
   Answer:
 
-  5. All files with extension ".c" that are in a folder named "src"
+  echo *[!aeiou]
+5. All files with extension ".c" that are in a folder named "src"
   Answer:
 
-  6. All files with extension ".c" under any folder
-  Answer:
+ ls *.c src/ 
+6. All files with extension ".c" under any folder
+  Answer: ?:(
+ 
 
 2. Write a command that would delete all files ending with a ".png" under any subdirectory.
 
-3. The command "ls -l" can take the name of a file or directory and show details on them. From within "wildcards" folder run "ls -l [A-Z]*". What was printed? Why?
+ Amswer: find . -type f -name *.png -delete
+3. The command "ls -l" can take the name of a file or directory and show details on them. From within "wildcards" folder run "ls -l [A-Z]*". What was printed? Why?
+
 
 4. Search in "man ls" for an option to show only directory data so the command "ls _____ D*" should print only the word Doc
 
```

Comment:
The switch is -d: Normally ls shows all files inside a directory when given, so `ls Doc` would not show Doc but a list of all files inside. With wildcards this behavior is a bit confusing, so we have -d that tells ls to show only folder details and not content. `ls -d D*` will just print Doc, and `ls -ld D*` will show the long list with Doc permissions and details